Makefile

```diff
@@ -208,56 +154,193 @@ MKDOCS = $(RUN) mkdocs
 mkd-%:
 	$(MKDOCS) $*
 
-PROJECT_FOLDERS = jsonldcontext jsonschema owl python rdf
-git-init-add: git-init git-add git-commit git-status
-git-init:
-	git init
-git-add: .cruft.json
-	git add \
-		*.md \
-		.cruft.json \
-		.github \
-		.gitignore \
-		CODE_OF_CONDUCT.md \
-		CONTRIBUTING.md \
-		LICENSE \
-		MAINTAINERS.md \
-		Makefile \
-		README.md \
-		RELEASE_NOTES_v7.7.2_to_v7.7.7.md \
-		about.yaml \
-		assets \
-		images \
-		mkdocs.yml \
-		nmdc_schema \
-		notebooks \
-		poetry.lock \
-		project.Makefile \
-		project/ \
-		pyproject.toml \
-		src/ \
-		tests \
-		utils
-	git add $(patsubst %, project/%, $(PROJECT_FOLDERS))
-
-git-commit:
-	git commit -m 'Initial commit' -a
-
-git-status:
-	git status
-
-# only necessary if setting up via cookiecutter
-.cruft.json:
-	echo "creating a stub for .cruft.json. IMPORTANT: setup via cruft not cookiecutter recommended!" ; \
-	touch $@
-
 clean:
 	rm -rf $(DEST)
 	rm -rf tmp
 	rm -rf docs/*.md
 	rm -rf docs/*.html
 
-include project.Makefile
+###########################################################
+# Schema Release Tracking
+###########################################################
+
+REPO  := microbiomedata/nmdc-schema
+FILE  := nmdc_schema/nmdc_materialized_patterns.yaml
+LATEST_RELEASE_TAG_FILE := local/latest_release_tag.txt
+LATEST_TAG_SCHEMA_FILE   := local/nmdc_schema_last_release.yaml
+
+# Get the tag that belongs to the latest (non-prerelease) GitHub release
+LATEST_TAG = $(shell curl -fsSL https://api.github.com/repos/$(REPO)/releases/latest | jq -r '.tag_name')
+LATEST_TAG_SCHEMA_URL := https://raw.githubusercontent.com/$(REPO)/$(LATEST_TAG)/$(FILE)
+
+.PHONY: $(LATEST_TAG_SCHEMA_FILE) examples-clean
+
+# Rule to fetch the schema file if local/nmdc_schema_last_release.yaml does not exist OR if there is a new release
+$(LATEST_TAG_SCHEMA_FILE): $(LATEST_RELEASE_TAG_FILE)
+	@echo "Checking for schema updates..."
+	@if [ ! -f $@ ]; then \
+		echo "Schema file does not exist. Creating it..."; \
+		curl -fsSL $(LATEST_TAG_SCHEMA_URL) -o $@; \
+		echo "$(LATEST_TAG)" > $(LATEST_RELEASE_TAG_FILE); \
+	elif [ "$$(cat $(LATEST_RELEASE_TAG_FILE))" != "$(LATEST_TAG)" ]; then \
+		echo "New release detected ($(LATEST_TAG)). Downloading schema..."; \
+		curl -fsSL $(LATEST_TAG_SCHEMA_URL) -o $@; \
+		echo "$(LATEST_TAG)" > $(LATEST_RELEASE_TAG_FILE); \
+	else \
+		echo "Local copy of schema is already up to date with release $(LATEST_TAG)."; \
+	fi
+
+# Rule to store the latest release tag locally
+$(LATEST_RELEASE_TAG_FILE):
+	@if [ -f $(LATEST_TAG_SCHEMA_FILE) ]; then \
+	echo "ERROR: Tag file is missing. Recreating release tag and removing local schema file..."; \
+		rm -f $(LATEST_TAG_SCHEMA_FILE); \
+	fi
+	@echo "Creating release tag file..."
+	@curl -fsSL $(LATEST_TAG_SCHEMA_URL) -o $(LATEST_TAG_SCHEMA_FILE)
+	@echo "$(LATEST_TAG)" > $@
+	@echo "Release tag file created with tag: $(LATEST_TAG)"
```

Comment:
I'm not sure I understand the purpose of these. Why would you download an `nmdc-schema` file from GitHub when you're working in the `nmdc-schema` repo?

_Edit_: I just realized that this wasn't really added, but moved from the now-deleted `project.Makefile`. I still don't understand the point of it, though 😁

Reply:
it goes along with the MongoDB dumps, so one can run validation over any dump with any schema, but specifically the last schema here/ If I was going to develop that any further, i would go by the schema version announced by the API that was going to be dumped. But you have convinced me to remove all of that.